### PR TITLE
MUMPFL-22, MUMPFL-23: Emergency Contact Info

### PR DESCRIPTION
**Pull request by jhanstra**
Works similarly to Local Address. 'Nuff said.

image

image

image

image

image

### CODE REVIEW

File: my-profile-webapp/src/main/webapp/my-app.js [hunk 1]
```diff
@@ -230,7 +294,7 @@
           //var contactInfoPromise = $http.get('/profile/samples/contact-info.json');
           var contactInfoPromise = $http.get('/profile/api/contactInfo.json');
           var basicInfoPromise = $http.get('/profile/samples/basic-info.json');
-          var emergencyInfoPromise = $http.get('/profile/samples/emergency-info.json');
+          var emergencyInfoPromise = $http.get('/profile/api/emergencyInfo.json');
```
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/my-app.js]: This can be removed

File: my-profile-webapp/src/main/webapp/partials/countries.html [hunk 2]
```diff
@@ -0,0 +1,251 @@
+<select id="countries" class="form-control" required ng-model="contact.addresses[0].country">
```
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/countries.html]: Did we mean to have that ng-model be only the first address in the array?

Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/countries.html]: I think if you changed it to address.country it should be fine.

File: my-profile-webapp/src/main/webapp/partials/emergency-info.html [hunk 3]
```diff
@@ -1,24 +1,57 @@
...
+<div class="portlet-frame" style='display: inline-block; width: 100%;'>
+    <portlet-header app-title="My Profile - Emergency Info" app-image="img/square.jpg"></portlet-header>
+
+    <div class="portlet-body">
+        <!-- App Navigation if Needed -->
+        <div class="inner-nav-container">
+            <ul class="inner-nav">
+                <li ><a href="#/main">My Profile</a></li>
+                <li class="active"><a href="#/emergencyInfo">Emergency Info</a></li>
+            </ul>
```
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/emergency-info.html]: Did we update this navigation on the My Profile main page?

File: my-profile-webapp/src/main/webapp/partials/emergency.html [hunk 4]
```diff
@@ -0,0 +1,79 @@
+<div ng-if='!contact.edit || contact.edit == false'>
+    <p>{{ contact.preferredName }}</p>
+    <p ng-repeat="line in contact.addresses[0].addressLines">{{ line }}</p>
+    <p>{{ contact.addresses[0].city }}, {{ contact.addresses[0].state }} {{ contact.addresses[0].postalCode }}</p>
```
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/emergency.html]: addresses[0]?

File: my-profile-webapp/src/main/webapp/partials/emergency.html [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+    <br>
+    <div ng-repeat="phone in contact.phoneNumbers">
+        <p>{{ phone.value }} ({{ phone.type }})</p>
+    </div>
+    <br>
+    <p ng-if="contact.relationship">Relationship: {{ contact.relationship }}</p>
+    <p ng-if="contact.languages[0]">First Language: {{ contact.languages[0].value }}</p>
+    <p ng-if="contact.languages[1]">Second Language: {{ contact.languages[1].value }}</p>
```
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/emergency.html]: are we hardcoding 2 langs?

File: my-profile-webapp/src/main/webapp/partials/main.html [hunk 6]
```diff
@@ -7,8 +7,7 @@
         <div class="inner-nav-container">
             <ul class="inner-nav">
                 <li class="active"><a href="#/default">My Profile</a></li>
-                <li><a href="#/local">Local Address</a></li>
```
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/main.html]: could prob drop this page if we are not going to use it.

File: my-profile-webapp/src/main/webapp/partials/emergency-info.html (hunk 3 above)
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/emergency-info.html]: we did

**Comment by timlevett** [comment]
If you add a second or 3rd phone how do you delete it? Same with other arrays

File: my-profile-webapp/src/main/webapp/partials/emergency.html (hunk 5 above)
Comment (apetro) [inline my-profile-webapp/src/main/webapp/partials/emergency.html]: Language is very likely to be revised to a free-text "Advice about communicating with this contact" kind of prompt, giving an example of "Speaks Spanish but not English".

Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/emergency.html]: There is no restrictions on what you can put in that box now. So that would work

**Comment by timlevett** [comment]
Overall looks great @jhanstra . Are we limiting them to 1 address per emergency contact?

**Comment by keirserrie** [comment]
Awesome. I have a couple of UX-ey nitpicks but will write stories for next sprint. :+1:

File: my-profile-webapp/src/main/webapp/partials/main.html (hunk 6 above)
Comment (jhanstra) [inline my-profile-webapp/src/main/webapp/partials/main.html]: Done

File: my-profile-webapp/src/main/webapp/partials/emergency.html (hunk 5 above)
Comment (jhanstra) [inline my-profile-webapp/src/main/webapp/partials/emergency.html]: I agree with @apetro's approach. And yea, if we went this way, we should un-hardcode Second Language, good catch.

File: my-profile-webapp/src/main/webapp/my-app.js (hunk 1 above)
Comment (jhanstra) [inline my-profile-webapp/src/main/webapp/my-app.js]: Done

File: my-profile-webapp/src/main/webapp/partials/countries.html (hunk 2 above)
Comment (jhanstra) [inline my-profile-webapp/src/main/webapp/partials/countries.html]: Yea I did mean to only use the first in the addresses array - do we really need to support multiple addresses for emergency contacts? Even one doesn't seem incredibly needed - in an emergency you're not going to mail someone something. 

We could revise the json structure to only have one address instead or an array, or we can just target the first element of the array. Thoughts?

Comment (jhanstra) [inline my-profile-webapp/src/main/webapp/partials/countries.html]: With the current json structure address.country wouldn't work - needs to be addresses[0].country right?

File: my-profile-webapp/src/main/webapp/partials/emergency.html (hunk 4 above)
Comment (jhanstra) [inline my-profile-webapp/src/main/webapp/partials/emergency.html]: same reason as above

File: my-profile-webapp/src/main/webapp/partials/countries.html (hunk 2 above)
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/countries.html]: We want to keep the same JSON object for emergency and local so we will keep it an array. Talk in the room says 1 address is cool for emergency.

**Comment by jhanstra** [comment]
@timlevett we should add support for deleting secondary items. We might want to make that a separate story cuz there's a decent amount of work to that.

I would like to limit to one address per emergency contact. But that's something to get feedback about from the stakeholders and come back to if we need to add multiple address support.

**Comment by timlevett** [comment]
@jhanstra agreed on secondary item.

File: my-profile-webapp/src/main/webapp/partials/emergency.html (hunk 4 above)
Comment (timlevett) [inline my-profile-webapp/src/main/webapp/partials/emergency.html]: agreed

**Comment by timlevett** [comment]
:+1:

**Pull request merged**